lib/segment/src/index/field_index/geo_hash.rs

```diff
@@ -67,22 +181,27 @@ fn sphere_lat(lat: f64) -> f64 {
 }
 
 /// Get neighbour geohash even from the other side of coordinates
-fn sphere_neighbor(hash_str: &str, direction: Direction) -> Result<GeoHash, GeohashError> {
-    let (coord, lon_err, lat_err) = decode(hash_str)?;
+fn sphere_neighbor(hash: GeoHash, direction: Direction) -> Result<GeoHash, GeohashError> {
+    let hash_str = SmolStr::from(hash);
+    let (coord, lon_err, lat_err) = decode(hash_str.as_str())?;
     let (dlat, dlng) = direction.to_tuple();
     let lon = sphere_lon(coord.x + 2f64 * lon_err.abs() * dlng);
     let lat = sphere_lat(coord.y + 2f64 * lat_err.abs() * dlat);
 
     let neighbor_coord = Coord { x: lon, y: lat };
-    encode(neighbor_coord, hash_str.len()).map(Into::into)
+    encode(neighbor_coord, hash_str.len())
+        .map(Into::<SmolStr>::into)
+        .map(Into::into)
 }
 
 pub fn encode_max_precision(lon: f64, lat: f64) -> Result<GeoHash, GeohashError> {
-    encode((lon, lat).into(), GEOHASH_MAX_LENGTH).map(Into::into)
+    encode((lon, lat).into(), GEOHASH_MAX_LENGTH)
+        .map(Into::<SmolStr>::into)
+        .map(Into::into)
 }
 
-pub fn geo_hash_to_box(geo_hash: &GeoHash) -> GeoBoundingBox {
-    let rectangle = decode_bbox(geo_hash).unwrap();
+pub fn geo_hash_to_box(geo_hash: GeoHash) -> GeoBoundingBox {
+    let rectangle = decode_bbox(SmolStr::from(geo_hash).as_str()).unwrap();
```

Comment:
And we might also benefit from a direct to `String` conversion, versus an intermediate `SmolStr`